### ISSUE TITLE
type error when overridden method uses `functools.cache` decorator

### ISSUE THREAD

**Issue by comex** [microsoft/pyright#3577]
**To Reproduce**
Type-checking this in strict mode:
```py
import functools

class Foo:
    def func(self) -> int:
        return 1

class Bar(Foo):
    @functools.cache
    def func(self) -> int:
        return 2
```
produces:
```
  /private/tmp/test.py:8:5 - error: "func" overrides method of same name in class "Foo" with incompatible type "_lru_cache_wrapper[int]" (reportIncompatibleMethodOverride)
```

**Expected behavior**
This should type-check, since the object returned by `functools.cache` mimics the original function.  It can be called with the same signature, and is compatible with use as a method (i.e. it is properly bound to a method object when `Bar` instances are created).

**VS Code extension or command-line**
Command-line tool, 1.1.254, typeshed 4967c8bc
